sky/packages/sky/lib/gestures/recognizer.dart

```diff
@@ -0,0 +1,141 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:sky' as sky;
+
+import 'package:sky/base/pointer_router.dart';
+import 'package:sky/gestures/arena.dart';
+import 'package:sky/gestures/constants.dart';
+
+abstract class GestureRecognizer extends GestureArenaMember {
+  GestureRecognizer({ PointerRouter router }) : _router = router;
+
+  PointerRouter _router;
+
+  final List<GestureArenaEntry> _entries = new List<GestureArenaEntry>();
+  final Set<int> _trackedPointers = new Set<int>();
+
+  void addPointer(sky.PointerEvent event);
+  void handleEvent(sky.PointerEvent event);
+  void didStopTrackingLastPointer();
```

Comment:
Add the following, to remind readers that there's more abstract methods to implement, and to narrow the signatures so since this class opines on the nature of keys:
```dart
  void acceptGesture(int key);
  void rejectGesture(int key);
```